### PR TITLE
Layout to-do list content in a simple two-column layout

### PR DESCRIPTION
**Pull request by jwalker625**
Adding components for the to-do list content.  Goal is a simple 2-column layout with the task list on the left and an action card on the right.

Result: 
image

### CODE REVIEW

File: src/ToDoDetail/ToDoDetail.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    return (
+      <div {...otherProps} className={classNames(className, "ToDoDetail")}>
+        <h2>Add New Task</h2>
+      </div>
+    );
+  }
+
+  if(!selectedTask) {
+    return (
+      <div {...otherProps} className={classNames(className, "ToDoDetail")}>
```
Comment (rs-lkroneman) [inline src/ToDoDetail/ToDoDetail.tsx]:
```suggestion
      <div {...otherProps} className={classNames(className, "ToDoDetail")}>
```

consider making this a component that receives children.

File: src/App/App.tsx [hunk 2]
```diff
@@ -1,10 +1,19 @@
+import React from 'react';
 import './App.css';
 import ToDoHeader from '../ToDoHeader';
+import ToDoList from '../ToDoList';
+import ToDoDetail from '../ToDoDetail';
 
 const App = () => {
+  const [selectedTask, setSelectedTask] = React.useState(null);
```
Comment (rs-lkroneman) [inline src/App/App.tsx]:
```suggestion
  const [selectedTask, setSelectedTask] = React.useState();
```

🗣️ can leave this as undefined as well if you're not planning to initialize it right away.

File: src/App/__tests__/App.test.tsx [hunk 3]
```diff
@@ -1,13 +1,53 @@
...
+  //   const user = userEvent.setup();
+  //   render(<App />);
+  //
+  //   const addNewTaskButton = await screen.findByRole('button', { name: '+ New Task' });
+  //   await user.click(addNewTaskButton);
+  //
+  //   const addNewTaskForm = await screen.findByText(/Add New Task/i);
+  //
+  //   expect(addNewTaskForm).toBeInTheDocument();
+  // });
```
Comment (rs-lkroneman) [inline src/App/__tests__/App.test.tsx:52]:
```suggestion
```

🗣️ typically commented code should be removed for a pull request.x

**Review by rs-lkroneman — APPROVED** [review]

File: src/ToDoDetail/ToDoDetail.tsx (hunk 1 above)
Comment (jwalker625) [reply src/ToDoDetail/ToDoDetail.tsx]: Resolved in 85d206b

File: src/App/App.tsx (hunk 2 above)
Comment (jwalker625) [reply src/App/App.tsx]: Whether I use `null` or `undefined`, I get a type error, `Type 'undefined' is not assignable to type '{ title: string; description: string; deadline?: Date | undefined; }'. `, unless I include a `| undefined` or `| null` in the type below.  Is there a better way of allowing `selectedTask` to be undefined?

```
type ToDoDetailProps = {
  isNewTask?: boolean,
  selectedTask: {
    title: string,
    description: string,
    deadline?: Date
  } | undefined
} & React.ComponentProps<'div'>;
```

**Pull request merged**